app/controllers/pieces_controller.rb

```diff
@@ -35,4 +43,16 @@ def valid_move?
       redirect_to game_path(@piece.game), alert: "This is not a valid move!"
     end
   end
+
+  def can_castle? 
+    piece = Piece.find(params[:piece_id])
+    rook = Piece.find(params[:rook_id])
+    if !piece.can_castle?(rook)
+      redirect_to game_path(piece.game), alert: "You can not Castle at this time!"
+    end
+
+    if piece.is_white? && piece.game.player_one != current_user || !piece.is_white? && piece.game.player_two != current_user
+      redirect_to game_path(piece.game), alert: "That is not your piece!"
```

Comment:
We already have validations for knowing if player is clicking on a piece that is not theirs. (see comment at top of this file)